protocol/streaming/full_node_streaming_manager.go

```diff
@@ -809,28 +889,38 @@ func (sm *FullNodeStreamingManagerImpl) StreamBatchUpdatesAfterFinalizeBlock(
 
 	finalizedFills, finalizedSubaccountUpdates := sm.getStagedEventsFromFinalizeBlock(ctx)
 
-	// TODO(CT-1190): Stream below in a single batch.
-	// Send orderbook updates to sync optimistic orderbook onchain state after FinalizeBlock.
-	sm.SendOrderbookUpdates(
+	orderbookStreamUpdates, orderbookClobPairIds := getStreamUpdatesFromOffchainUpdates(
 		orderBookUpdatesToSyncLocalOpsQueue,
 		uint32(ctx.BlockHeight()),
 		ctx.ExecMode(),
 	)
 
-	// Send finalized fills from FinalizeBlock.
-	sm.SendOrderbookFillUpdates(
+	fillStreamUpdates, fillClobPairIds := sm.getStreamUpdatesForOrderbookFills(
 		finalizedFills,
 		uint32(ctx.BlockHeight()),
 		ctx.ExecMode(),
 		perpetualIdToClobPairId,
 	)
 
-	// Send finalized subaccount updates from FinalizeBlock.
-	sm.SendFinalizedSubaccountUpdates(
+	sm.Lock()
+	defer sm.Unlock()
```

Comment:
for my understanding - why do we need to lock before `getStreamUpdatesForSubaccountUpdates`?

Reply:
We don't; just wanted to lock before the 1st flush. I re-ordered the code blocks a bit to optimize